docs/model/product-search-via-elasticsearch.md

```diff
@@ -57,18 +65,48 @@ But if we simplify, we can say that the search term is searched in attributes an
 * short description - match anywhere
 * description - match anywhere
 
-The searched fields and their priority are defined directly in the `Shopsys\FrameworkBundle\Model\Product\Search\ProductElasticsearchRepository::createQuery()` function.
+The searched fields and their priority are defined directly in the `Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery::search()` method.  
+This method can be easily changed by extending `FilterQuery` class.
 
 If you want to improve searching, you can learn more in [Elasticsearch analysis](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis.html).
 
+### Product filtering
+
+Products can be by default filtered by price, flags, brand, parameters and in stock availability.
+
+Behavior of filter is defined in `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade::getPaginatedProductsInCategory()` and
+`Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade::getPaginatedProductsForSearch()` methods.  
+Each internally uses their own factory method `createProducts*FilterQuery` to create `Shopsys\FrameworkBundle\Model\Product\Search\FilterQuery` object that represents the query for Elasticsearch.
+
+Elasticsearch return sorted list of product IDs and products itself are loaded from PostgreSQL.
+
+Aggregation numbers are counted with help of Elasticsearch too thanks to methods `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade::getProductFilterCountDataInCategory` and
+`Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainElasticFacade::getProductFilterCountDataForSearch`.
+
+List of choices (exact parameters, brands, flags) is loaded from PostgreSQL as there is no benefit from loading them from Elasticsearch.
+
+## Implementation
+
+`ProductController` and `SearchController` uses `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface` to get filtered products.
+
+Currently there are two implementations of `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface`:
+ - `ProductOnCurrentDomainFacade`
+    - filters data through SQL
+    - slower than Elasticsearch, but on the other hand can be used easily on more complex pricing models (for example exact price is calculated with SQL function)
+ - `ProductOnCurrentDomainElasticFacade`
+    - filters data through Elasticsearch
+    - much faster than filtering through SQL and remain fast independently of the number of selected filters
+
+Class used can be easily changed in `services.yml` file in your project.
```

Comment:
Do you think this is enough or should we have information about what you need to change in the `services.yml` file to change implementations?
```suggestion
You can choose which implementation is used in `services.yml` file in your project.
```